src/input/input_python.rs

```diff
@@ -271,15 +273,35 @@ impl<'a> Input<'a> for PyAny {
         }
     }
 
+    #[cfg(not(PyPy))]
     fn lax_list(&'a self) -> ValResult<GenericListLike<'a>> {
         if let Ok(list) = self.cast_as::<PyList>() {
             Ok(list.into())
         } else if let Ok(tuple) = self.cast_as::<PyTuple>() {
             Ok(tuple.into())
-        } else if let Ok(set) = self.cast_as::<PySet>() {
-            Ok(set.into())
-        } else if let Ok(frozen_set) = self.cast_as::<PyFrozenSet>() {
-            Ok(frozen_set.into())
+        } else if let Ok(iterator) = self.cast_as::<PyIterator>() {
+            let tuple = PyTuple::new(self.py(), iterator.iter()?.flatten().collect::<Vec<_>>());
```

Comment:
is there a particular reason you're using `PyTuple` here?

Reply:
tuple instantiation is faster than list. So I suppose it's the same for pytuple

Reply:
See https://github.com/samuelcolvin/rust-bench/commit/3651309db7b5d4e8e8ee6196a1b385350543e4a1.

I give up on computers, the benchmarks mostly indicate that tuples are a little faster than lists:

```
running 2 tests
test instantiation_list         ... bench:       1,109 ns/iter (+/- 11,654)
test instantiation_tuple        ... bench:         911 ns/iter (+/- 66)

test result: ok. 0 passed; 0 failed; 0 ignored; 2 measured; 24 filtered out; finished in 14.06s
```

(note the massive range on the list case).

Sometime the results are reversed:

```
running 2 tests
test instantiation_list         ... bench:         794 ns/iter (+/- 884)
test instantiation_tuple        ... bench:         904 ns/iter (+/- 28)

test result: ok. 0 passed; 0 failed; 0 ignored; 2 measured; 24 filtered out; finished in 4.27s
```

Reply:
But the other thing is that if there's no validator, `PyList` can be passed straight back, so regardless of whether instantiating a tuple is faster, validation will be massively faster if we use `PyList` here in the case of no validator.

Reply:
ahhh sorry for `lax_list` yes it's supposed to be `PyList` directly! This is a mistake with bad copy pasting. For the rest we can keep `PyTuple` right?